resizing_hashtable/r_hashtables.py

```diff
@@ -62,61 +73,95 @@ def hash_table_insert(hash_table, key, value):
 def hash_table_remove(hash_table, key):
    # same here we want to get the hashed key
     index = hash(key,  hash_table.capacity)
-    # Take the hash key and modulus with size(capacity) to get the index
-
+    current_pair = hash_table.storage[index]
+    prev_pair = None
     # Take the index of the storage and verify if its set to None
-    if hash_table.storage[index] is not None:
-        # if the hash storage is not none. Then set it to None
-        hash_table.storage[index] = None
-    else:
-        # If it is none then we can send a message to Say There is not a key in the table
-        print(f'{key} is not in the hash table. So it can not be removed')
+    if current_pair is not None:
+       # if it not none then check if the key is = to the storage key
+        while current_pair is not None and current_pair.key != key:
+            prev_pair = current_pair
+            current_pair = current_pair.next
+
+        if prev_pair == None and current_pair.key == key:
+            hash_table.storage[index] = None
+            hash_table.count -= 1
+        elif current_pair is None:
+            print(f'Error! {key} not found')
+        else:
+            prev_pair.next = None
 
+    else:
+        print(f'Error! {key} not found')
 
 # '''
 # Fill this in.
 
 # Should return None if the key is not found.
 # '''
+
+
 def hash_table_retrieve(hash_table, key):
-       # same here we want to get the hashed key
+    print("I am Running")
+    # same here we want to get the hashed key
     index = hash(key,  hash_table.capacity)
+    current_pair = hash_table.storage[index]
+    print(f'The Retrieve, is in Index {index} has {key}')
+    # Take the index of the storage and verify if its set to None
+    if current_pair is not None:
 
-    if hash_table.storage[index] is None:
-        return None
+        while current_pair is not None and current_pair.key != key:
+            current_pair = current_pair.next
+        if current_pair == None:
+            print(f'Error {key} not found')
+        else:
+            return current_pair.value
+
+    else:
+        print(f'Error {key} not found')
 
-    return hash_table.storage[index].value
 # '''
 # Fill this in
 # '''
 
 
 def hash_table_resize(hash_table):
-    new_hash = HashTable(hash_table.capacity * 2)
+    new_hash_table = HashTable(hash_table.capacity * 2)
 
-    for i in range(hash_table.capacity):
-        print(hash_table.storage[i].key)
-        new_hash.storage[i] = hash_table.storage[i]
-    return new_hash
+    for x in range(hash_table.count):
+        # new_hash_table.storage[index] = hash_table.storage[index]
+        current_pair = hash_table.storage[x]
+        while current_pair is not None:
+            hash_table_insert(
+                new_hash_table, current_pair.key, current_pair.value)
+            current_pair = current_pair.next
+
+    return new_hash_table
 
+    new_hash = HashTable(hash_table.capacity * 2)
```

Comment:
I believe this block of code will not get executed after the return statement on line 38.